allauth/socialaccount/providers/dataporten/tests.py

```diff
@@ -0,0 +1,67 @@
+from allauth.socialaccount.tests import OAuth2TestsMixin
+from allauth.tests import MockedResponse, TestCase
+
+from .provider import DataportenProvider
+
+
+class DataportenTest(OAuth2TestsMixin, TestCase):
+    provider_id = DataportenProvider.id
+
+    def setUp(self):
+        super(DataportenTest, self).setUp()
+        self.mock_data = {
+            'userid': '76a7a061-3c55-430d-8ee0-6f82ec42501f',
+            'userid_sec': ['feide:andreas@uninett.no'],
+            'name': 'Andreas \u00c5kre Solberg',
+            'email': 'andreas.solberg@uninett.no',
+            'profilephoto': 'p:a3019954-902f-45a3-b4ee-bca7b48ab507',
+            'groups': [{}],
+        }
+
+    def get_login_response_json(self, with_refresh_token=True):
+        rt = ''
+        if with_refresh_token:
+            rt = ',"refresh_token": "testrf"'
+        return '''{
+            "access_token":"testac",
+            "expires_in":3600,
+            "scope": "userid profile groups"
+            %s
```

Comment:
I can't find a single mention of refresh tokens in the [Dataporten documentation](https://docs.dataporten.no). But as I understand it, it is meant to be included here according to the oAuth2 specification. The default implementation seems to have almost the same behaviour.